### PR TITLE
Clarify full nodes limitations

### PR DESCRIPTION
**Pull request by nuke-web3**
Based on discussions with @bkchr on our technical channels, this PR:
- updates limitations of full nodes implementations (for now)
- references `smoldot` as a light node of interest

@bkchr, please do review this and let me know if anything needs to be amended. 🙏🏼

### CODE REVIEW

**Comment by emresurmeli** [comment]
Please make sure to rebase master and check to see your changes are reflected locally before continuing work. We merged a new version of Docusaurus and have ran basic checks to see if it works, but could run into minor issues.

File: docs/maintain-sync.md [hunk 1]
```diff
@@ -28,25 +28,29 @@ was, or which extrinsics resulted in a certain state change are fast operations
...
-behind, a pruned full node will not be able to sync to the network.
-
 Archive nodes are used by utilities that need past information - like block explorers, council
 scanners, discussion platforms like [Polkassembly](https://polkassembly.io), and others. They need
-to be able to look at past on-chain data. Full nodes are used by everyone else - they allow you to
-read the current state of the chain and to submit transactions directly to the chain without relying
-on a centralized infrastructure provider.
+to be able to look at past on-chain data.
+
+A **full node** is _pruned_, meaning it discards all information older than a specific number of
```
Comment (bkchr) [inline docs/maintain-sync.md]: We only discard finalized blocks.

File: docs/maintain-sync.md [hunk 2]
```diff
@@ -28,25 +28,29 @@ was, or which extrinsics resulted in a certain state change are fast operations
...
+blocks (256 blocks by default), and the genesis block. A node that is pruned this way requires
```
Comment (bkchr) [inline docs/maintain-sync.md]: Why do you mention the genesis block explicit?

**Review by bkchr — COMMENTED** [review]
Looks good from my side. Besides the nitpicks.

File: docs/maintain-sync.md (hunk 2 above)
Comment (nuke-web3) [reply docs/maintain-sync.md]: Was here in the wiki before, and my intent is to imply you need a chainspec that includes this block (or the information to produce it?), so all nodes retain this at least.

File: docs/maintain-sync.md (hunk 1 above)
Comment (nuke-web3) [reply docs/maintain-sync.md]:
```suggestion
A **full node** is _pruned_: it discards all finalized blocks older than a configurable
```

File: docs/maintain-sync.md (hunk 2 above)
Comment (nuke-web3) [reply docs/maintain-sync.md]:
```suggestion
number (256 blocks by default), and the genesis block. A node that is pruned this way requires
```

Comment (bkchr) [reply docs/maintain-sync.md]: But I read this as it also discards the genesis block :D But I'm no native speaker

Comment (nuke-web3) [reply docs/maintain-sync.md]: yap. That is a good point. Updated.

**Comment by nuke-web3** [comment]
@emresurmeli is the wiki in a good state to rebase this on for a merge? There are no merge conflicts, so I think it's OK to use (text only changed). I think it is ready otherwise.

**Comment by nuke-web3** [comment]
Rebased and ready to merge (with positive review)

**Review by salmad3 — APPROVED** [review]

**Pull request merged**